app/controllers/api/v1/users_controller.rb

```diff
@@ -1,20 +1,24 @@
 class Api::V1::UsersController < ApplicationController
   skip_before_action :authenticate_request,
-                     :email_verified?,
+                     :is_email_verified?,
                      only: [:create_patient]
-  before_action :restrict_user, only: %i[index show_user]
-  before_action :set_user, only: [:show_user]
+  before_action :restrict_user, only: %i[show_users show_user]
+  before_action :restrict_patient, only: [:show_patients]
+  before_action :set_user, only: %i[show_user destroy_user]
 
-  def index
-    @users =
-      User.all.select do |user|
-        (user.roles.first == patient_role || user.roles.first == doctor_role)
-      end
+  def show_users
+    @users = User.all.each.select { |user| user.roles.first != admin_role }
     render json: @users, status: :ok
   end
 
-  def show_current_user
-    render json: @current_user, status: :ok
+  def show_doctors
+    @users = User.all.each.select { |user| user.roles.first == doctor_role }
```

Comment:
I think we can put this code in initializer method:
```ruby
@users = User.all.each.select { |user| user.roles.first }
```
so it can be reused by other method and no need to query every time we call this method.